src/core/ExtensionOverrides.ts

```diff
@@ -0,0 +1,40 @@
+import { File } from './File';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class ExtensionOverrides {
+  public overrides: string[];
+
+  constructor(overrides: string[]) {
+    this.overrides = [];
+    overrides.forEach((override) => this.add(override));
+  }
+
+  private isValid(override: string) {
+    return (typeof override === 'string' && override.indexOf('.') === 0);
+  }
+
+  public add(override: string) {
+    if (this.isValid(override)) {
+      this.overrides.push(override);
+    }
+  }
+
+  public setOverrideFileInfo(file: File): string {
+    if (this.overrides.length === 0) {
+      return null;
+    }
+
+    const fileInfo = path.parse(file.info.absPath);
+
+    for (let overrideExtension of this.overrides) {
+      const overridePath = path.resolve(fileInfo.dir, `${fileInfo.name}${overrideExtension}`);
+
+      if (overrideExtension.indexOf(fileInfo.ext) > -1 && fs.existsSync(overridePath)) {
+        file.absPath = file.info.absPath = overridePath;
```

Comment:
i don't think we should override absPath ( my bad,  I told you that )
we should map it in the context instead for the watcher

Reply:
Using HMR_FILE_REQUIRED
https://github.com/fuse-box/fuse-box/blob/8586d90baac5326f849aadf8aca82377a7e33764/src/plugins/stylesheet/CSSplugin.ts#L132-L137